lib/gollum-lib/filter/yaml.rb

```diff
@@ -0,0 +1,26 @@
+require 'yaml'
+# Extract YAML frontmatter from data and build metadata table. 
+
+class Gollum::Filter::YAML < Gollum::Filter
+  
+  # Regexp thanks to jekyll
+  YAML_FRONT_MATTER_REGEXP = %r!\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)!m
+
+  def extract(data)
+    data.gsub!(YAML_FRONT_MATTER_REGEXP) do
+      @markup.metadata ||= {}
+      begin
+        frontmatter = ::YAML.load(Regexp.last_match[1])
+        frontmatter = frontmatter.respond_to?(:to_h) ? frontmatter.to_h : Hash.new(frontmatter)
```

Comment:
What is the use case for frontmatter that needs to be coerced into a hash via `Hash.new`?